test/lib/agreement.js

```diff
@@ -44,6 +44,16 @@ contract('Delivery Interface', function(accounts) {
     assert.equal(await token.balanceOf(agreementContract.address), 600);
   });
 
+  it('should catch if not enough tokens', async () => {
+    agreement = new Agreement(offer.address, 10000000000000, token.address, AgreementArtifacts, Token);
+    try {
+      var agreementContract = await agreement.initiateAgreement();
+      assert(false);
+    } catch (e) {
+      assert(true);
+    }
+  });
+
```

Comment:
We can now do it much better, we can use OpenZeppelin (we can add it to a project with npm, https://github.com/OpenZeppelin/zeppelin-solidity) and use expectThrow (https://github.com/OpenZeppelin/zeppelin-solidity/blob/master/test/helpers/expectThrow.js):

`await expectThrow(() => await agreement.initiateAgreement()); `